subnet/license.go

```diff
@@ -188,11 +220,22 @@ func (lv *LicenseValidator) ValidateEnterpriseLicense(acceptedPlans []string) (*
 		}
 	}
 	if !accepted {
-		return nil, fmt.Errorf("this tool/service is not available to %s customers", li.Plan)
+		return nil, fmt.Errorf("this software is only available for license plans %v", strings.Join(acceptedPlans, ", "))
 	}
 
-	if li.ExpiresAt.Before(time.Now()) && li.IsTrial {
-		return nil, fmt.Errorf("trial license has expired on %v", li.ExpiresAt)
+	if li.ExpiresAt.Before(time.Now()) {
+		if li.IsTrial || li.Plan == "TRIAL" {
+			// no grace period for trial
+			return nil, fmt.Errorf("trial license has expired on %v", li.ExpiresAt)
+		}
+		// expired, within grace period. print warning.
+		fmt.Printf("license has expired on %v, renew immediately to avoid outage.", li.ExpiresAt)
```

Comment:
```suggestion
		fmt.Printf("license has expired on %v, renew immediately to avoid outage.", li.ExpiresAt)
```

All this logging must be avoided here, the library must just generates errors or reports for the caller to get the right set of information to show.

It should be printing this on its own, this can cause problems for logging systems under each of these projects.

Reply:
Removed all logging.